### PR TITLE
[BT-1808] dxskyap switch to requests and support for TLS v1.1

### PR DESCRIPTION
**Pull request by careyrp**
This is a fix for our dxskytap library to continue to work with skytap. Skytap will be removing support for TLS 1.0 in June. The following changes configure our dxskytap client to use TLS 1.2

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+requests == 2.10
+pexpect == 2.4
+urllib3 == 1.20
+pyOpenSSL == 0.14
+cryptography == 1.3.4
+idna == 2.0.0
+
```
Comment (cchristous) [inline requirements.txt]: you can set the contents of this file to be `-e .` to use the versions in `setup.py`. also, you should not specify specific versions for a library. otherwise, the users of the library can't effectively use the library as well as other libraries due to version conflicts.

File: tox.ini [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
-envlist = py26,py27,py34
+envlist = py27,py34,py35
 
 [testenv]
+commands=nosetests
 deps=
    nose
    coverage
-commands=nosetests
+   -rrequirements.txt
```
Comment (cchristous) [inline tox.ini]: I believe you can accomplish the same thing by adding a `tests_require` section to the `setup.py`

File: dxskytap/connect.py [hunk 3]
```diff
@@ -31,46 +31,62 @@
 
 Exceptions:
  - SkytapException
- - TimeoutException
  - NoResponseException
 '''
 
+import urllib3.contrib.pyopenssl
+urllib3.contrib.pyopenssl.inject_into_urllib3()
```
Comment (cchristous) [inline dxskytap/connect.py]: why is this needed rather than simply installing `requests[secure]`?

Comment (careyrp) [inline dxskytap/connect.py]: Because the default backend for urllib3 has to be compiled in. Meaning if the upgrade openssl, in order to see the changes in python you have to recompile and reinstall python core. This is an alternative backend for urllib3 that uses the pyopenssl library. pyopenssl is an alternative library to the core ssl library that provides better control and integration with openssl.

File: requirements.txt (hunk 1 above)
Comment (careyrp) [inline requirements.txt]: other projects will not be using the requirements.txt file for dxskytap. They will have their own requirements.txt file. In that situation the other project just needs to meet the setup.py.  This requirements.txt is only used and needed when testing dxskytap.

File: tox.ini (hunk 2 above)
Comment (careyrp) [inline tox.ini]: there is a test_requires that includes these libraries. Tox will not automatically run the setup.py. Instead of setup.py, I set it up to run requirements.txt to lock the versions.

Comment (cchristous) [inline tox.ini]: if you change the command here to `python setup.py nosetests`, it will automatically install the dependencies in `tests_require` before running the tests

File: requirements.txt (hunk 1 above)
Comment (cchristous) [inline requirements.txt]: Maybe change the name to `test_requirements.txt`, if you even still need it

File: tox.ini [hunk 4]
```diff
@@ -1,9 +1,9 @@
 # content of: tox.ini , put in same dir as setup.py
 [tox]
-envlist = py26,py27,py34
+envlist = py27,py34,py35
```
Comment (careyrp) [inline tox.ini]: The changes made to use TLS 1.2 are not compatible with python 2.6. Removed python 2.6 from list of test environments.

**Comment by babineaum** [comment]
:shipit:
